### ISSUE TITLE
Observations and suggestions about the mod

### ISSUE THREAD

**Issue by wormzjl** [GlobbyPotato/Rockhounding_Chemistry#34]
Hello, thank you for making such an interesting mod. I work in the field of chemical engineering and it's great to see something about chemistry. I've made some tests and here are some observations and suggestions:

First of all some small bugs:
In air cooling, 1 cu of refined air gives 100 cu of cooled air, which does not correspond to JEI entry.
On page 135 of the guidebook, the schema of the crystal pulling crucible is misplaced.

And some questions and suggestions:
1. Is it possible to use fluids from other mods as gases in this mod, I would like to use IC2 superheated steam in certain custom recipes, but it seems that there is no method of connecting normal fluids reservoirs with gas tanks (gas pipeline pump does no connect to normal tanks). If this is the case, is it possible to add a conversion method?

2. In the begining of the manual it would be helpful to put a couple of screenshots about the connection types (blue, orange, yello...) since it can be easily missed by new players.

3. For the heat exchanger, the usable coolant is not very clear. I've tried to use water but there must be other fluids I guess? 
At industrial scale, the heat exchanger is often used to recover heat from hot fluids to heat up cold fluids for energy saving purpose, maybe it can be put to more use in future applications?

4. Also a suggestion regarding the realism of catalyst preparation: normally a catalyst is made from a solid porous support (alumina, titanium dioxide or silica, etc) and a metal salt. The support is immerged in such a metal salt solution, then calcinated and activated with hydrogen. The used catalyst can also be regenerated via various methods.

5. A gas void would be helpful in case you want to get rid of excess gases, especially when distillating air.

6. Do you have plans of adding some sort of standard chemical devices such as continuously stirred tank reactor, autoclave, fluidized bed reactor...etc.

7. Is it possible to have localization support for the guidebook?

Thank you again!

**Comment by GlobbyPotato** [comment]
Hi, thanks for the feedback, not many people find this mod even useful so that's really apreciated and gave me useful hints for fixing. Let's see to sort out the points:

About the bugs you are right, there was an unnoticed mistake making one or more machines possibly not performing correctly. It should work now. Thanks for reporting this issue. Fixed also the Crucible scheme that was not alligned.

1) In the latest upload I added a Transposer that should do this kind of integrations via CraftTweaker. Still experimental though, so not full explaination is provided yet.
With it I swapped the sulfuric acid variants from Mekanism and Galacticraft into the Rockhounding one to avoid removing and re-adding all the recipes using it and still using all the supplies.
About gases it still needs some deeper testing because I'm expert at all with some of the bigger mods and not really sure of how other mods handle their gases and transportation. From what I have set for now, if gases are marked as "gaseous" and can travel into normal fluid pipes and tanks, the Transposer should be able to pick them. The script uses the same gas for input and output, or use any other variant to swap. This way the Gas Pipeline can read them.
I currently found a bug in the script so it may not work untill the next upload.

2) I added a description about the color code in the tech concept chapter, will try to improve this part. Problem I found is people don't read guides or criticize that it's too big.

3)The heat exchanging started like something else so it is a wide approximation of the concept behind it, to make it suitable for the resources process/consume purpose.
Currently you can use any fluid that is not hotter than ambient temperature (water is 300K by Forge). The lower is the temperature of the coolant the less power it requires and less coolant is consumed. 
I'll make a fix to make sure the temperature of the fluids is shown in JEI tooltips since it seems to be enforced only by a Wawla for what I see.
In one of the next uploads I was planning to force the coolant temperature to be as well lower than the input temperature, hoping to don't make it too stressing for players.
I thought about the heat recovery function, did not implemented it because there wasn't yet a feature requiring this, indeed only the air refrigeration is actually done so far.

4) The catalyt part is also approximated because it may create a loophole since I need to make sure the mod can run on it's own without any other external mod. Exactly for their preparation methods it can happen that to make the catalyst to extract elements XYZ you may need elements XYZ to prepare the catalyst. Not even sure the catalyst I added are really accurate for the same reason.
The regeneration part is interesting indeed, but the problem I faced is that players want linear automation, so the catalyst is supposed to enter the machine and consume so the hopper can put another one in place. The only regeneration method I can think of is a "mending" process, like adding another block in the structures using catalysts which unlocks the regeneration by supplying the necessary resources.

5) Every tank and vessel has already a Void button to clean the content, it is the button with the sort of asterisk. In air distillation only the gas bulbs don't have one because the content is fixed and it must be anyway extracted to holders. You probably would be able to disable the single rare gases to be extracted? If it's so it could be done, not all the rare gases have an application yet, they are up for grab for pack implementation.

6) I already got requests for the CSTR. To add it, I need to see first of all how to implement it in the current system or it will be a random machine with no usage. Eventually it could alleviate the use of the Lab Oven.

7) For this I don't really know. I use an external mod where I write an XML file. Probably I'd need a cloned file with the content in another language. But the content may change over time making all difficult.

Thanks

**Comment by wormzjl** [comment]
Thank you for your reply. Just curious, do you have a roadmap for the mod?

Regarding the CSTR, back in the age of 1.7.10 there was a mod called IHL which introduced a very interesting and realistic mechanism: you have to introduce the reagents stoichiometrically into the reactor, otherwise it will be filled up with a single reagent after a few cycles. It also have a redstone-controlled dosing pump which allows precise control of injected fluid amounts. This makes players actually need to think about how to automate a reaction instead doing a+b=c like most mods. 

About the popularity of this mod, I think the major thing blocking the access of this mod is documentation and visibility, reading through the book might be difficult for many players and I spent half a day trying to figure out how every machine works. I'd suggest making some tutorial and update videos like nuclearcraft (https://www.youtube.com/channel/UC-df90SAvYXk-FClmntlFhw/videos) since many things are changed between versions, and post stuffs like showcases to ftb reddit (https://reddit.com/r/feedthebeast/). Also a video walkthrough about early to mid age progression will be helpful.

**Comment by GlobbyPotato** [comment]
Not really a roadmap. The main goal was to make the complex factory to get the late game resources that are not minable. The path is pretty ended, other then bugfixes or ways to bridge better some passages. For the rest I'm probably dropping the additional features I started in 1.10 because other mods already covered them better and with easier passages, So not worth the time spent to code and bugfix them with the lower skills.

I'm not sure I want to go this deeper to reprerent a reaction like that mod, the intention was to find a balance between the representation of the process and the assembling puzzle. What I reproduced here has been already labeled as tedious by pack players as far as I've seen, so not willing to make things even more complicated or scholastic.

Unfortunately all I can provide is the guidebook or the wiki and answers to questions when made, I don't have the tools and the hardware to make videos. I lately play from my old backup 32bit pc, so not good at all.
